src/xr/xr-input-source.js

```diff
@@ -299,6 +325,19 @@ class XrInputSource extends EventHandler {
         return this._localRotation;
     }
 
+    /**
+     * @function
+     * @name XrInputSource#getLinearVelocity
+     * @description Get the linear velocity (units per second) of input source if it is handheld ({@link XrInputSource#grip} is true). Otherwise it will return null.
+     * @returns {Vec3|null} The world space linear velocity of handheld input source.
+     */
+    getLinearVelocity() {
```

Comment:
We already have `RigidBodyComponent#linearVelocity`. Should this be a property too?

Reply:
Fair enough, but whole XR APIs uses function notation. Also it is only a get function, there is no way to set linear velocity as it is sourced from underlying APIs.

Reply:
Then it would just be a getter property. No requirement to have a setter. Functions are fine for `getPosition`/`setPosition` etc to mirror the function on `GraphNode` but generally, we should use properties where it makes sense.

Reply:
If we want to use properties, then all other methods would need to be changed for consistency reasons. Preferring a consistent design for WebXR APIs.

Reply:
The WebXR APIs shouldn't define their own API style. There should be an engine-wide style. If the WebXR APIs don't adhere to that, we can gradually migrate and alias to something that is consistent with the overall engine.

Reply:
I've used `get*` style which is in line with `Entity` and other APIs. And as mentioned before, this is not settable property like in rigidbody's case.
This particular method can be changed to match very specific case of rigidbody.
Then should we change: `getOrigin` and `getDirection` of `XrInputSource` to be similar to `Ray`, but again mentioning: origin and direction of input source cannot be set by developer, and provided by an API. Also they do call a function inside to update transform, just like `GraphNode.getPosition` does.

And regarding `getPosition` and `getRotation`, should we change them to simply: `position` and `rotation`?

To be honest, it is already not consistent within engine, but it is very clear that if it is a property - then it can be set. In XR case, those methods cannot be set, they only provide data.

Reply:
There are a large number of gettable but not settable properties in the API. Just check out `GraphNode` for example:

https://developer.playcanvas.com/en/api/pc.GraphNode.html#children

Read-only properties are 100% OK.

`getPosition` and `getRotation` could be aliased to `position` and `rotation` maybe. I think it's a little complicated and needs some serious thought.

TBH, we should probably improve `Ray` and have the `XrInputSource` provide a `Ray` instead of separate `origin` and `direction`.

Reply:
We can definitely start a discussion and do WebXR-wide refactor, with such changes.